packages/components/src/components/select-native/select-native.jsx

```diff
@@ -15,9 +15,27 @@ const getDisplayText = (list_items, value) => {
     return '';
 };
 
-const SelectNativeOptions = ({ list_items, should_hide_disabled_options, use_text }) => {
+const SelectNativeOptions = ({ label, list_items, should_hide_disabled_options, use_text }) => {
     const options = should_hide_disabled_options ? list_items.filter(opt => !opt.disabled) : list_items;
-
+    const grouped = options.reduce((r, a) => {
+        r[a.group] = [...(r[a.group] || []), a];
+        return r;
+    }, {});
+    const keys = Object.keys(grouped);
+    const values = Object.values(grouped);
+    if (label === 'Asset') {
+        return keys.map((option, i) => {
+            return (
+                <optgroup key={option} label={option}>
+                    {values[i].map(value => (
+                        <option key={value.value} value={use_text ? value.text : value.value}>
+                            {value.nativepicker_text || value.text}
+                        </option>
+                    ))}
+                </optgroup>
+            );
+        });
+    }
```

Comment:
-  Let’s check on ‘Assets’ first before grouping and use push() instead of destructuring and creating an array on every iteration as it is more memory consuming.

```
const groups = {};
options.forEach(option => {
  const group = option.group;
   if (!groups[group]) {
     groups[group] = [];
   }
   groups[group].push(option);
 })
```

-  let's use more descriptive names for variables -> change `keys` for `group_names`
-  instead of assigning Object.values to a new variable let's access them using group_name inside map loop  it would decrease memory usage
    
```
group_names.map((group_name, group_idx) => {
      const group = groups[group_name]
    })
```

Reply:
Thanks for good advices. Removed check by 'Assets' and use push, Object.values method and renamed variable.